ferrocene/doc/user-manual/src/rustfmt/usage.rst

```diff
@@ -0,0 +1,58 @@
+.. SPDX-License-Identifier: MIT OR Apache-2.0
+   SPDX-FileCopyrightText: The Ferrocene Developers
+
+Using ``rustfmt``
+=================
+
+``rustfmt`` formats Rust code or checks if Rust code is well formatted.
+
+``rustfmt <file>...``
+---------------------
+
+To format one or more files, just pass the paths to ``rusfmt``. This will format the passed files and all of their child modules.
+
+Assuming following project structure, ``rustfmt main.rs`` will format the whole project, while ``rustfmt mod1.rs`` will format ``mod1.rs``, ``mod1/mod10.rs`` and ``mod1/mod11.rs``, but neither ``mod2.rs`` nor ``mod2/mod20.rs``.
+
+.. code-block::
+
+   $ tree src/
+   src/
+   ├── main.rs
+   ├── mod1
+   │   ├── mod10.rs
+   │   └── mod11.rs
+   ├── mod1.rs
+   ├── mod2
+   │   └── mod20.rs
+   └── mod2.rs
+
+
+``rustfmt --check <file>...``
+-----------------------------
+
+If the ``--check`` option is passed ``rusfmt`` checks if the files are well formatted. It reports what it would format, but will not modify any files. For example:
```

Comment:
```suggestion
If the ``--check`` option is used, ``rusfmt`` checks if the files are well formatted. It reports what it would format, but will not modify any files. For example:
```